lib/posix-process/signal/sigsuspend.c

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <stddef.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <uk/syscall.h>
+
+#include "signal.h"
+#include "sigset.h"
+
+UK_SYSCALL_R_DEFINE(int, rt_sigsuspend,
+		    const uk_sigset_t *, mask,
+		    size_t, sigsetsize)
+{
+	struct posix_thread *pthread;
+	uk_sigset_t saved_mask;
+
+	pthread = tid2pthread(uk_gettid());
+
+	UK_ASSERT(pthread);
+
```

Comment:
Let's also throw `EFAULT` as per manual if `mask` is invalid.

Reply:
Due to reasons discussed in an earlier comment we can only limit this check to `if (!mask)`. Is that what you meant?